mcs/class/corlib/Test/System.Threading/WaitHandleTest.cs

```diff
@@ -395,6 +395,30 @@ public void InterrupedWaitOne ()
 			}
 		}
 
+		[Test]
+		public void InterruptedWaitOneWithTimeout ()
+		{
+			using (ManualResetEvent mre = new ManualResetEvent (false))
+			using (ManualResetEvent ready = new ManualResetEvent (false)) {
+				var thread = new Thread (() => {
+					ready.Set ();
+					mre.WaitOne (1000);
+				});
+
+				thread.Start ();
+				ready.WaitOne ();
+
+				Thread.Sleep (10); // wait a bit so we enter mre.WaitOne
+
+				DateTime start = DateTime.Now;
+				while (DateTime.Now < start.AddMilliseconds (500)) {
+					GC.Collect (1);
+				}
+
+				Assert.IsTrue (thread.Join (1000), "#2");
```

Comment:
Is there a possibility of making the test flaky given that this waits exactly as long as the mre thread, so we might timeout here before the test finishes? Can we wait it a bit longer here?